distutils/sysconfig.py

```diff
@@ -449,6 +449,14 @@ def get_config_vars(*args):
     global _config_vars
     if _config_vars is None:
         _config_vars = sysconfig.get_config_vars().copy()
+        if os.name == 'nt':
+            # See https://github.com/pypa/distutils/issues/130
+            _config_vars['EXT_SUFFIX'] = _imp.extension_suffixes()[0]
+            if not IS_PYPY:
+                # For backward compatibility, see issue19555
+                SO = _config_vars.get('EXT_SUFFIX')
+                if SO is not None:
+                    _config_vars['SO'] = SO
```

Comment:
I didn't see anything in the report about SO, and in fact the Python bug fix removes the support for SO. I don't disagree it's important, but probably it should be handled differently. Do we know under what conditions this additional variable is needed?

Reply:
Looking at [the bug](https://bugs.python.org/issue19555), the SO var has been deprecated since 2013. Let's just omit it unless someone complains.

Reply:
Fixed in 74dc5363b3.

Reply:
There is a test that checks SO is equal to EXT_SUFFIX, that's why I added it. Maybe we can remove that test?

Reply:
So much tech debt. :)

Yes, I see how now these things are coupled, because [sysconfig sets SO to match EXT_SUFFIX](https://github.com/python/cpython/blob/785cc6770588de087d09e89a69110af2542be208/Lib/sysconfig.py#L671-L673), SO needs to be updated when EXT_SUFFIX is updated.

Reply:
What about PyPy? I guess it must have different logic for setting that value.

Reply:
PyPy doesn't set `SO` at all. They set `EXT_SUFFIX` to the correct value since pypy3.5 I think. https://github.com/mozillazg/pypy/commit/775a03237c4a233cc2afcf3bc6df2f863d58d4b5

Reply:
PyPy [has the same implementation](https://foss.heptapod.net/pypy/pypy/-/blob/60ca933d0bb52ef4144591424d2ca4686483e594/lib-python/3/sysconfig.py#L592-594), so I'm thinking PyPy shouldn't be treated differently.

Reply:
Agreed

Reply:
55da5cbe37 restores SO support without PyPy as a factor.